src/services/notification.service.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const { EventEmitter } = require('events');
+
+const EVENTBUS_EVENT_NAME = 'message';
+
+const WalletEventMap = {
+  state: 'wallet:state',
+  'new-tx': 'wallet:new-tx',
+  'udpate-tx': 'wallet:udpate-tx',
+};
+
+const ConnectionEventMap = {
+  state: 'conn:state',
+  'wallet-update': 'conn:wallet-update',
+  'best-block-update': 'conn:best-block-update',
+  'wallet-load-partial-update': 'conn:wallet-load-partial-update',
+};
+
+const AllEvents = [...Object.values(WalletEventMap), ...Object.values(ConnectionEventMap)];
+
+// TODO: methods and properties to handle the wallet events
+class HathorEvents extends EventEmitter {
+  subscribeHathorWallet(walletId, hwallet) {
+    // Wallet events
+
+    // state
+    hwallet.on('state', state => {
+      const type = WalletEventMap.state;
+      const data = { type, walletId, data: state };
+
+      this.emit(type, data);
+      this.emit(EVENTBUS_EVENT_NAME, data);
+    });
+
+    // new-tx
+    hwallet.on('new-tx', tx => {
+      // XXX: Check if it's from the wallet?
```

Comment:
The wallet will emit this event only for txs from it.

Reply:
Forgot to remove my annotations